shared/src/main/scala/squants/energy/MolarEnergy.scala

```diff
@@ -0,0 +1,48 @@
+package squants.energy
+
+import squants.mass.{ChemicalAmount, Moles}
+import squants.{AbstractQuantityNumeric, Dimension, PrimaryUnit, Quantity, SiUnit, UnitConverter, UnitOfMeasure}
+
+/**
+  *
+  * @author Nicolas Vinuesa
+  * @since 1.4
+  *
+  * @param value Double
```

Comment:
I mean, I'm not very familiar with `MolarEnergy`, perhaps we could add a link to wikipedia?